Service/Validate/Regex.php

```diff
@@ -13,60 +13,29 @@
 
 class Regex implements ValidationServiceInterface
 {
-    /**
-     * @var ConfigInterface
-     */
-    private $vatNumberConfig;
-
     /**
      * @var array
      */
-    private $regexMap;
+    private $vatPatternMap = [];
 
     /**
      * Regex constructor.
      * @param ConfigInterface $vatNumberConfig
      */
     public function __construct(ConfigInterface $vatNumberConfig)
     {
-        $this->vatNumberConfig = $vatNumberConfig;
+        // get the patterns from the configuration
```

Comment:
I usually try to prevent executing code in the constructor, but this looks good to me.

Reply:
Agreed, I never do business logic. or anything that might throw errors.